src/Microsoft.Extensions.Configuration/ConfigurationRoot.cs

```diff
@@ -24,6 +24,12 @@ public ConfigurationRoot(IList<IConfigurationProvider> providers)
             _providers = providers;
         }
 
+        public string Key => string.Empty;
+
+        public string Path => string.Empty;
+
+        public string Value { get; set; }
```

Comment:
If Value makes no sense on the root, can we just make the setter throw and always return null?  Isn't that safe and reasonable?